services/src/main/java/org/keycloak/services/resources/IdentityBrokerService.java

```diff
@@ -670,38 +670,20 @@ private Response afterFirstBrokerLogin(ClientSessionCode<AuthenticationSessionMo
                 federatedUser.grantRole(readTokenRole);
             }
 
-            // Add federated identity link here
```

Comment:
Moved everything around creating the user to the step that creates the user so it's not split into multiple requests. Not sure if this is just a bit to naive approach or not